src/main/java/org/commcare/formplayer/sandbox/SqlStorageWrapper.java

```diff
@@ -71,12 +47,16 @@ private void compareRunnable(String tag, Runnable sqliteOperation, Runnable post
     private <RESULT> RESULT compareCallable(String tag, Callable<RESULT> sqliteOperation, Callable<RESULT> postgresOperation) {
         RESULT result = null;
         try {
-            Timer sqliteTimer = meterRegistry.timer("timer.sqlite." + tag);
-            result = sqliteTimer.recordCallable(sqliteOperation);
+            long start = System.currentTimeMillis();
+            result = sqliteOperation.call();
+            long sqliteTime = System.currentTimeMillis() - start;
+
+            start = System.currentTimeMillis();
+            postgresOperation.call();
+            long postgresTime = System.currentTimeMillis() - start;
 
-            Timer postgresTimer = meterRegistry.timer("timer.postgres." + tag);
-            postgresTimer.recordCallable(postgresOperation);
-            // TODO compare result with sqlite
+            Timer diffTimer = meterRegistry.timer("storage.timing." + tag);
+            diffTimer.record(postgresTime - sqliteTime, TimeUnit.MILLISECONDS);
```

Comment:
what made you change the method of capturing the timing?

Reply:
I had a chat with Kevin about it. He mentioned that capturing the difference would be more important so that we can tie each 2 requests.